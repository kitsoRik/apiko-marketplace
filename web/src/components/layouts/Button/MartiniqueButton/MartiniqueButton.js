import React from 'react';

import "./MartiniqueButton.scss";

const MartiniqueButton = ({ className, value, disabled = false, ...props }) => {
    return (
        <button
            className={`button-martinique ${className ?? ""}`}
            disabled={disabled}
            {...props}
        >
            {value}
        </button>
    )
};

export default MartiniqueButton;