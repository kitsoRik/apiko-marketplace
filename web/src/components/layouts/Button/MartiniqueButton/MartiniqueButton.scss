.button-martinique {
    padding: 20px 50px;

    border: none;
    
    color: #fff;
    background-color: #3E3961;

    &:hover {
        background-color: #3E5991;
    }

    &:active {
        background-color: #3E4971;
    }
}