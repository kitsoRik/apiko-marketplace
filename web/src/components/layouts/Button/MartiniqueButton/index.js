import MartiniqueButton from "./MartiniqueButton";

export default MartiniqueButton;