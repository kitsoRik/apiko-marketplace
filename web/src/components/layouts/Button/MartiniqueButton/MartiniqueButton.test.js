import React from "react";
import { create } from "react-test-renderer";
import MartiniqueButton from "./MartiniqueButton";
import { mount, shallow } from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";

configure({ adapter: new Adapter() });

describe("MartiniqueButton", () => {
	it("should render without value", () => {
		const component = create(<MartiniqueButton />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with value", () => {
		const component = create(<MartiniqueButton value="My custom value" />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render disabled", () => {
		const component = create(
			<MartiniqueButton value={"My custom calue"} disabled={true} />
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should have class", () => {
		const component = mount(<MartiniqueButton />);
		expect(component.find("button").hasClass("button-martinique")).toEqual(
			true
		);
	});

	it("should skip click event", () => {
		const onClick = jest.fn();
		const component = mount(
			<MartiniqueButton onClick={onClick} disabled={true} />
		);
		component.find("button").simulate("click");
		expect(onClick).toBeCalledTimes(0);
	});

	it("should emit click event", () => {
		const onClick = jest.fn();
		const component = mount(<MartiniqueButton onClick={onClick} />);
		const button = component.find("button");

		button.simulate("click");
		expect(onClick).toBeCalledTimes(1);
		button.simulate("click");
		expect(onClick).toBeCalledTimes(2);
	});
});
