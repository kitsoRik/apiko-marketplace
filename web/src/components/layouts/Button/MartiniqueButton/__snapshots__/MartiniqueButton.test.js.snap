// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MartiniqueButton should render disabled 1`] = `
<button
  className="button-martinique "
  disabled={true}
>
  My custom calue
</button>
`;

exports[`MartiniqueButton should render with value 1`] = `
<button
  className="button-martinique "
  disabled={false}
>
  My custom value
</button>
`;

exports[`MartiniqueButton should render without value 1`] = `
<button
  className="button-martinique "
  disabled={false}
/>
`;
